### PR TITLE
chatall: init at 1.54.83

### PR DESCRIPTION
**Pull request by paveloom**
## Description of changes

https://github.com/sunner/ChatALL

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/ch/chatall/package.nix [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      desktopName = "ChatALL";
+      exec = "chatall";
+      icon = "chatall";
+      comment = meta.description;
+      categories = [ "Utility" "Chat" ];
+    })
+  ];
+
+  postPatch = ''
+    ${jq}/bin/jq \
```
Comment (FliegendeWurst) [inline pkgs/by-name/ch/chatall/package.nix:52]:
```suggestion
    jq \
```

File: pkgs/by-name/ch/chatall/package.nix [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    rev = "v${version}";
+    hash = "sha256-Gi08TdD6Wl7PUPmNJdjtaMsEdZB2CU1X2I+dhPv1O5U=";
+  };
+
+  npmDepsHash = "sha256-pDuIoAsUEhwJ7TLpslE7VRhoXD3f0X5ipVjtrJmGwPI=";
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    icoutils
+    makeWrapper
```
Comment (FliegendeWurst) [inline pkgs/by-name/ch/chatall/package.nix:31]:
```suggestion
    makeWrapper
    jq
    moreutils # for sponge
```

File: pkgs/by-name/ch/chatall/package.nix [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      comment = meta.description;
+      categories = [ "Utility" "Chat" ];
+    })
+  ];
+
+  postPatch = ''
+    ${jq}/bin/jq \
+      '.main = "dist_electron/bundled/index.js" |
+      del(.dependencies."electron-builder")' \
+      package.json | ${moreutils}/bin/sponge package.json
```
Comment (FliegendeWurst) [inline pkgs/by-name/ch/chatall/package.nix:55]:
```suggestion
      package.json | sponge package.json
```
Using ${package}/bin/... will not work when cross-compiling.

File: pkgs/by-name/ch/chatall/package.nix [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share/lib/chatall
+    cp -r dist/*-unpacked/{locales,resources{,.pak}} $out/share/lib/chatall
+
+    mkdir -p $out/share/icons/hicolor/256x256/apps
+    icotool -x -i 1 -o $out/share/icons/hicolor/256x256/apps/chatall.png src/assets/icon.ico
+
+    makeWrapper ${electron}/bin/electron $out/bin/chatall \
```
Comment (FliegendeWurst) [inline pkgs/by-name/ch/chatall/package.nix:77]:
```suggestion
    makeWrapper ${lib.getExe electron} $out/bin/chatall \
```

**Review by FliegendeWurst — COMMENTED** [review]
Package builds and launches.
It outputs `update-electron-app config looks good; aborting updates since app is in development mode`, but I don't know whether the app being in "development mode" is a problem for other functionality (disabling the auto-updater is good).

**Comment by paveloom** [comment]
Forced it to think it's packaged and in production (similar to how it's done in [`youtube-music`](https://github.com/NixOS/nixpkgs/blob/8ab2f09522d5aefabf1100b3f59a3bde628cb2b4/pkgs/applications/audio/youtube-music/default.nix#L74-L75)), so the message is gone now.

**Review by FliegendeWurst — APPROVED** [review]
Builds and launches successfully on x86_64-linux (when cherry-picked onto current master).

I'm surprised the "Help" menu is empty though. Is that expected for this program?

New output is `Electron's autoUpdater does not support the 'linux' platform`, which is good (for now).

**Comment by paveloom** [comment]
> I'm surprised the "Help" menu is empty though. Is that expected for this program?

I don't know the reasons why it's like that, but I just checked their AppImage, and it's the same there.

**Comment by paveloom** [comment]
If you run into errors containing `Skia shader compilation error`, run `rm -rf ~/.config/chatall/GPUCache`.

**Pull request closed without merging**